src/types.js

```diff
@@ -24,16 +34,42 @@ export type OfflineAction = {
   }
 };
 
+export type NetInfo = {
+  isConnectionExpensive: ?boolean,
+  reach: string
+};
+
+export type OfflineStatusChangeAction = {
+  type: typeof OFFLINE_STATUS_CHANGED,
+  payload: {
+    online: boolean,
+    netInfo: NetInfo
```

Comment:
NetInfo is an optional value shouldn't this be `?NetInfo`

Reply:
You are right. I just haven't used the library on the browser so I missed that. It should be `netInfo?: NetInfo`
Should `netinfo` be optional also in the following block?
```
export type OfflineState = {
  busy: boolean,
  lastTransaction: number,
  online: boolean,
  outbox: Outbox,
  netInfo: NetInfo,
  retryCount: number,
  retryScheduled: boolean
};
```

Only when all files will be 100% flow typed we will really be able to spot 100% types bugs

Reply:
Agreed. Yes that's also optional.